lib/practice_exercises.rb

```diff
@@ -1,13 +1,48 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string.downcase!
+  string_array = string.split('')
+  allowed_characters = ('a'..'z').to_a
+  string_array.select! { |letter| allowed_characters.include?(letter) }
+  i = 0
+
+  while i <= string_array.length && !string_array[i].nil?
+    return false if string_array.first != string_array.last
+
+    string_array.pop
+    string_array.shift
+    i += 1
+  end
+
+  return true
+
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(n)
 def longest_prefix(strings)
```

Comment:
You have 2 loops, one which loops through the list of strings, n times.

The other loops through the length of the 1st word, m times.

Since they are nested this is O(nm) time complexity.

However since you are also sorting the strings this is O(nlogn + nm), do you need to sort?